src/coreclr/vm/methodtable.h

```diff
@@ -2998,10 +3029,11 @@ public :
             MethodTable* pMT;
         };
 
-        static void* operator new(size_t size, TargetMethodTable targetMT)
+        static void* operator new(size_t size, TargetMethodTable targetMT, MethodDataComputeOptions computeOptions)
         {
-            _ASSERTE(size <= GetObjectSize(targetMT.pMT));
-            return ::operator new(GetObjectSize(targetMT.pMT));
+            _ASSERTE(computeOptions != MethodDataComputeOptions::CacheOnly);
```

Comment:
Can you comment why this assumption is appropriate?

Reply:
Comment added.